app/js/checkAnswers.js

```diff
@@ -1,28 +1,33 @@
 const questionAmount = 30// amount of questions
 
-document.querySelector('#finish').addEventListener('click', function(e) {
-    var unanswered = questionAnswered()
-    if (unanswered == false) {
+document.querySelector('#finish').addEventListener('click', function() {
+    let unanswered = questionAnswered()
+    if (unanswered == false && !Object.values(flaggedQuestions).includes(true)) {
         showResults()
-    } else {
-        document.getElementById('modal-title').textContent = `You have ${unanswered.length} unanswered questions.`
+    }
+    else {
+        let flagList = document.getElementById('flag-list')
+        let unansweredList = document.getElementById('unanswered-list')
+        document.getElementById('modal-title').textContent = 'You have ' + unanswered.length + ' unanswered questions.'
+        flagList.innerHTML = ''
+        unansweredList.innerHTML = ''
+        Object.values(flaggedQuestions).forEach(function(isQuestionFlagged, qId) {
+            if (isQuestionFlagged) {
+                flagList.innerHTML += "<li>" + (qId + 1) +"</li>"
+            }
+        })
+        unanswered.forEach(function (isQuestionUnanswered) {
+            unansweredList.innerHTML += "<li>" + isQuestionUnanswered +"</li>"
```

Comment:
isQuestionUnanswered is not what this variable contains, it contains the ID of unanswered questions.

Reply:
Changed in 2ccbd9a